pkg/gcp/credentialsconfig.go

```diff
@@ -105,18 +104,26 @@ func getCredentialsConfigFromSecret(secret *corev1.Secret) (*CredentialsConfig,
 			return nil, fmt.Errorf("could not get credentials config from %q field: %w", ServiceAccountJSONField, err)
 		}
 		if credentialsConfig.ProjectID == "" {
-			credentialsConfig.ProjectID = string(secret.Data["projectID"])
+			credentialsConfig.ProjectID = string(secret.Data[ProjectIDField])
 		}
 		return credentialsConfig, nil
 	}
 
+	// Secrets other than shoot's cloudprovider does not have the `credentialsConfig` field set.
+	// Let's try to set it from the `config` key.
+	if _, ok := secret.Data[CredentialsConfigField]; !ok {
+		if err := SetWorkloadIdentityFeatures(secret.Data, WorkloadIdentityMountPath); err != nil {
+			return nil, fmt.Errorf("could not set workload identity features: %w", err)
+		}
+	}
```

Comment:
Do we need to modify the original secret? I find it counterintuitive that a function called `getCredentialsConfigFromSecret` can mutate the data. Is it possible to copy the secret data and work with that for the sake of constructing a `credentialsConfig`?